org.idempiere.webservices/WEB-INF/src/org/compiere/model/MWebServiceType.java

```diff
@@ -286,5 +286,45 @@ public X_WS_WebServiceFieldInput getFieldInput(String colName){
 		
 		return m_inputFieldMap.get(colName);
 	}
+
+	public static void insertParameters (Properties ctx, MWebServiceType ws, String trxName) {
+
+		if (ws.getWS_WebService().getValue().equals("ModelADService")) {
+			String method = ws.getWS_WebServiceMethod().getValue();
+
+			if (method.equals("getList")) {
+				addWsParameter(ctx, ws, "AD_Reference_ID", X_WS_WebService_Para.PARAMETERTYPE_Free, "", trxName);
+			}
+			else if (method.equals("runProcess")) {
+				addWsParameter(ctx, ws, "AD_Process_ID", X_WS_WebService_Para.PARAMETERTYPE_Constant, "", trxName); // can't fill it as the process is unknown
+				addWsParameter(ctx, ws, "AD_Menu_ID", X_WS_WebService_Para.PARAMETERTYPE_Constant, "0", trxName);
+				addWsParameter(ctx, ws, "AD_Record_ID", X_WS_WebService_Para.PARAMETERTYPE_Free, "", trxName);
+			}
+			else {
+				String value = "";
+				if (method.equals("createData")) value = "Create";
+				if (method.equals("deleteData")) value = "Delete";
```

Comment:
here and the next three lines is better to use an else if - that avoids checking for a condition that you already know is false

Reply:
done